OpenXmlPowerTools/DocumentBuilder.cs

```diff
@@ -1636,79 +1845,313 @@ private static void AddToIgnorable(XElement root, string v)
+                    itemList.Add(itemId);
+            foreach (CustomXmlPart customXmlPart in sourceDocument.MainDocumentPart.CustomXmlParts)
+            {
+                OpenXmlPart propertyPart = customXmlPart
+                    .Parts
+                    .Select(p => p.OpenXmlPart)
+                    .Where(p => p.ContentType == "application/vnd.openxmlformats-officedocument.customXmlProperties+xml")
+                    .FirstOrDefault();
+                if (propertyPart != null)
+                {
+                    XDocument propertyPartDoc = propertyPart.GetXDocument();
+                    if (itemList.Contains(propertyPartDoc.Root.Attribute(DS.itemID).Value))
+                    {
+                        CustomXmlPart newPart = newDocument.MainDocumentPart.AddCustomXmlPart(customXmlPart.ContentType);
+                        newPart.GetXDocument().Add(customXmlPart.GetXDocument().Root);
+                        foreach (OpenXmlPart propPart in customXmlPart.Parts.Select(p => p.OpenXmlPart))
+                        {
+                            CustomXmlPropertiesPart newPropPart = newPart.AddNewPart<CustomXmlPropertiesPart>();
+                            newPropPart.GetXDocument().Add(propPart.GetXDocument().Root);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static Dictionary<XName, XName[]> RelationshipMarkup = null;
+
+        private static void UpdateContent(IEnumerable<XElement> newContent, XName elementToModify, string oldRid, string newRid)
+        {
+            foreach (var attributeName in RelationshipMarkup[elementToModify])
+            {
+                var elementsToUpdate = newContent
+                    .Descendants(elementToModify)
+                    .Where(e => (string)e.Attribute(attributeName) == oldRid);
+                foreach (var element in elementsToUpdate)
+                    element.Attribute(attributeName).Value = newRid;
+            }
+        }
+
+        private static void AddRelationships(OpenXmlPart oldPart, OpenXmlPart newPart, IEnumerable<XElement> newContent)
+        {
+            var relevantElements = newContent.DescendantsAndSelf()
+                .Where(d => RelationshipMarkup.ContainsKey(d.Name) &&
+                    d.Attributes().Any(a => RelationshipMarkup[d.Name].Contains(a.Name)))
+                .ToList();
+            foreach (var e in relevantElements)
+            {
+                if (e.Name == W.hyperlink)
+                {
+                    string relId = (string)e.Attribute(R.id);
+                    if (string.IsNullOrEmpty(relId))
+                        continue;
+                    var tempHyperlink = newPart.HyperlinkRelationships.FirstOrDefault(h => h.Id == relId);
+                    if (tempHyperlink != null)
+                        continue;
+                    Guid g = Guid.NewGuid();
+                    string newRid = "R" + g.ToString().Replace("-", "");
```

Comment:
per https://docs.microsoft.com/en-us/dotnet/api/system.guid.tostring, you can just do (with string interpolation): `$"R{g:N}`. String concatentation is generally a bad idea as it creates multiple copies of strings, which is the most common way to overuse memory. In this case `+` is probably not too bad, but string interpolation uses `FormattableString` that's a bit smarter in generating the final string

Reply:
Made the change.